src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs

```diff
@@ -24,23 +24,27 @@ namespace OpenTelemetry.Instrumentation.GrpcNetClient.Implementation
     internal class GrpcClientDiagnosticListener : ListenerHandler
     {
         public const string RequestCustomPropertyName = "OTel.GrpcHandler.Request";
+        private readonly GrpcClientInstrumentationOptions options;
 
         private readonly ActivitySourceAdapter activitySource;
         private readonly PropertyFetcher startRequestFetcher = new PropertyFetcher("Request");
 
-        public GrpcClientDiagnosticListener(ActivitySourceAdapter activitySource)
+        public GrpcClientDiagnosticListener(ActivitySourceAdapter activitySource, GrpcClientInstrumentationOptions options)
             : base("Grpc.Net.Client")
         {
             if (activitySource == null)
             {
                 throw new ArgumentNullException(nameof(activitySource));
             }
 
+            this.options = options;
             this.activitySource = activitySource;
         }
 
         public override void OnStartActivity(Activity activity, object payload)
         {
+            SuppressInstrumentationScope.Begin(this.options.SuppressInstrumentation);
```

Comment:
`Begin` returns a scope that should be disposed at some point. Is this happening somehow?

Also, can we change this to the following to avoid allocating a scope?
```csharp
if (this.options.SuppressInstrumentation) SuppressInstrumentationScope.Begin()
```

Reply:
`Dispose` is not dis-allocating any object. It just resets the slot to previous value. I think call to `dispose` is not required here. Collected heap snapshot and verified `SuppressInstrumentationScope` getting cleared up after stop activity. Please let me know if I miss something here.

Reply:
I think this is sound when AsyncLocal/logical call context is the backing store. Though we might run into issues if the the underlying ContextSlotType is changed to ThreadLocal. Not sure this use case is actually possible today, though.

Reply:
@reyang It may make sense to refactor the RuntimeContext stuff a bit to ensure that this is not possible. That is, I think we'll always want an `AsyncLocal` based store for the `SuppressInstrumentationScope`. I don't think we'd want people to change the `ContextSlotType` via the API to something else and then have `SuppressInstrumentationScope` adopt whatever the user set it to. Thoughts?

Reply:
> It just resets the slot to previous value.

@rajkumar-rangaraj Isn't that the point though? Let's say this Grpc call is part of some other process the user is executing. If we don't reset the flag, it stays set and flows onward, no? Could that lead to accidental suppression of subsequent things?

Reply:
Makes sense, I will get this covered.